### PR TITLE
[GH-27] new endpoint to uncoolify

### PR DESCRIPTION
**Pull request by johncoleman83**
regarding issue #27 
- updated client with new endpoint.
- updated server with new endpoint
- added server tests for uncoolify service
- updated demo request url to include the user input and to have an option for both endpoints
- updated docs to reflect new endpoint
- Left 2 uncoolify options unfinished

### CODE REVIEW

File: views/index.html [hunk 1]
```diff
@@ -21,14 +21,16 @@ <h1 style="color:white;">
...
+          <button id='coolify-submit' type="submit">coolify</button>
+          <button id='uncoolify-submit' type="submit">uncoolify</button>
         </form>
       </div>
       <div class="content" uk-scrollspy="cls: uk-animation-slide-bottom-small; target: > div, div > .uk-card, .uk-card > .uk-card; delay: 100; repeat: true">
           <div class="uk-child-width-1-1@m uk-text-center uk-padding" uk-grid>
             <div>
                 <div class="uk-card json-card uk-card-secondary uk-card-body">
                     <span id="demo-url" style="color: black; background: white;" class="uk-padding-small uk-width-1-1" >
+                      submit input for demo
```
Comment (johncoleman83) [inline views/index.html]: This `span` will now be updated based off of this function: `updateDemoUrl()`

File: views/index.html [hunk 2]
```diff
@@ -21,14 +21,16 @@ <h1 style="color:white;">
         </h1>
         <form uk-sticky="animation: uk-animation-slide-top; offset: 10; show-on-up: true;">
           <input id="mah-nam" type="text" maxlength="100" placeholder="aye, sir!" required>
-          <button type="submit">coolify</button>
+          <button id='coolify-submit' type="submit">coolify</button>
```
Comment (johncoleman83) [inline views/index.html]: there are 2 unique functions defined in `public/client.js`, which are executed on `click` for each button.

**Comment by johncoleman83** [comment]
just updated tests due to a small bug, but they are passing now.

**Comment by mohitkyadav** [comment]
@johncoleman83 thanks for the PR, there's a small bug though, 
when uncooling `4y3_51r` should give `4y3_51r` but it actually gives `aye slr`, that's because whaile cooling we replace both `i`, and `l` with `1`. Any ideas how to solve this?

**Comment by johncoleman83** [comment]
ah, yeah I see what you mean... the only suggestion I have for that is to not have `i` and `l` both get encoded to the cool version of `1`.  I see the numbers as closes to Leet, `1337`, so maybe keel `l` as `1` and make `i` either `!` or keep `i` unchanged.  This will fix the decoding bug.

**Comment by johncoleman83** [comment]
Just fixed 2 bugs with underscores (oops)...  Want me to make sure no dictionaries have duplicate encodings such as both `i` and `l` mapping to `1`?

**Comment by mohitkyadav** [comment]
Please, proceed.

**Comment by mohitkyadav** [comment]
Can you also fix the bug with I, replace I with !.

**Comment by johncoleman83** [comment]
Do you mean that the coolify should coolify `L` (lowercase) -> `!` and uncoolify should uncoolify `!` -> `L` (lowercase) . and then therefore keep `i` as it is?

**Comment by johncoleman83** [comment]
Oh I see you mean `i` -> `!`  I agree, will do that.

**Comment by mohitkyadav** [comment]
No, coolify `i` with !, And l with 1 for l same goes for capital I and L

**Comment by johncoleman83** [comment]
Okay thanks.  I kept `l` (lowercase `L`) and `L` the same.  However, I realized that your demo placeholder is `aye, sir!` and it has both `i` and `!` in it, and in this case both those would also have the same encoding!  So to resolve that I made `i` and upside down exclamation mark, `¡`; It looks  a lot like `i` but it is actually different and will resolve all this confusion.  I added tests for this as well.

**Pull request merged**